app/src/main/kotlin/xyz/block/augurref/service/MempoolCollector.kt

```diff
@@ -71,6 +72,78 @@ class MempoolCollector(
     return latestFeeEstimate.get()
   }
 
+  fun getFeeEstimateForDateRange(
+    start_date: LocalDateTime,
+    end_date: LocalDateTime,
+    interval: Int,
+  ): List<FeeEstimate>? {
+    // Ensure start_date is not after end_date
+    if (start_date.isAfter(end_date)) {
+      logger.warn("Start date is after end date")
+      return null
+    }
+
+    // Initialize result list to store fee estimates
+    val estimates = mutableListOf<FeeEstimate>()
+    var currentDate = start_date
+
+    // Iterate through the date range with the given interval
+    while (!currentDate.isAfter(end_date)) {
+      logger.debug("Processing fee estimate for $currentDate")
+      val feeEstimate = getFeeEstimateForDate(currentDate)
+
+      // If fee estimate is non-null and has estimates, merge them
+      if (feeEstimate != null && feeEstimate.estimates.isNotEmpty()) {
+        // estimates.putAll(feeEstimate.estimates as Map<Int, BlockTarget>)
+        estimates.add(feeEstimate)
+      } else {
+        logger.debug("No valid fee estimate for $currentDate")
+      }
+
+      // Increment currentDate by interval (in minutes)
+      currentDate = currentDate.plusSeconds(interval.toLong())
+    }
+
+    // Return null if no estimates were collected
+    if (estimates.isEmpty()) {
+      logger.warn("No fee estimates collected for the date range")
+      return null
+    }
+
+    // Return a FeeEstimate with the aggregated estimates
+    return estimates
+  }
+
+  /**
+   * Get the fee estimate for specific date
+   */
+  fun getFeeEstimateForDate(dateTimeUTC: LocalDateTime): FeeEstimate? {
+    val dateTime = dateTimeUTC
+      .atZone(ZoneId.of("UTC")) // Interpret as UTC
```

Comment:
the logic to handle timezones is overly complex, what do you think about just using UTC timestamps instead?

```
// Single historical estimate
GET /historical_fee?timestamp=1723128639

// Date range with interval in seconds
GET /historical_fees?start_timestamp=1723128639&end_timestamp=1723214639&interval=3600
```